test/rules/typedef-whitespace/both/nospace/test.ts.lint

```diff
@@ -0,0 +1,815 @@
+                         ~          [expected nospace after colon in call-signature]
+        return "some name";
+    }
+
+    public set name(a   : string)   : void {}
+                       ~                        [expected nospace before colon in parameter]
+                         ~                      [expected nospace after colon in parameter]
+                                   ~            [expected nospace before colon in call-signature]
+                                     ~          [expected nospace after colon in call-signature]
+
+    public shemp(
+        a   : number,
+           ~            [expected nospace before colon in parameter]
+             ~          [expected nospace after colon in parameter]
+        b   : number
+           ~            [expected nospace before colon in parameter]
+             ~          [expected nospace after colon in parameter]
+    )   : void {}
+       ~            [expected nospace before colon in call-signature]
+         ~          [expected nospace after colon in call-signature]
+}
+
+class WithPreceedingSpacesWithSubsequentSpacesClass {
+    [index   :   number]   :   string
+            ~                          [expected nospace before colon in parameter]
+              ~                        [expected nospace after colon in parameter]
+                          ~            [expected nospace before colon in index-signature]
+                            ~          [expected nospace after colon in index-signature]
+
+    Prop   :   string = "some property";
+          ~            [expected nospace before colon in property-declaration]
+            ~          [expected nospace after colon in property-declaration]
+
+    public get name()   :   string {
+                       ~            [expected nospace before colon in call-signature]
+                         ~          [expected nospace after colon in call-signature]
+        return "some name";
+    }
+
+    public set name(a   :   string)   :   void {}
+                       ~                          [expected nospace before colon in parameter]
+                         ~                        [expected nospace after colon in parameter]
+                                     ~            [expected nospace before colon in call-signature]
+                                       ~          [expected nospace after colon in call-signature]
+
+    public shemp(
+        a   :   number,
+           ~            [expected nospace before colon in parameter]
+             ~          [expected nospace after colon in parameter]
+        b   :   number
+           ~            [expected nospace before colon in parameter]
+             ~          [expected nospace after colon in parameter]
+    )   :   void {}
+       ~            [expected nospace before colon in call-signature]
+         ~          [expected nospace after colon in call-signature]
+}
+
+// endregion
```

Comment:
Cool, this is fine then. I have some ideas for possible tweaks to the test system for the future which might change things anyways, so I'm good with leaving this as is :+1: